src/abstractitensornetwork.jl

```diff
@@ -483,3 +606,83 @@ function visualize(
   end
   return visualize(Vector{ITensor}(tn), args...; vertex_labels, kwargs...)
 end
+
+# 
+# Link dimensions
+# 
+
+function maxlinkdim(tn::AbstractITensorNetwork)
+  md = 1
+  for e in edges(tn)
+    md = max(md, linkdim(tn, e))
+  end
+  return md
+end
+
+function linkdim(tn::AbstractITensorNetwork, edge::Pair)
+  return linkdim(tn, edgetype(tn)(edge))
+end
+
+function linkdim(tn::AbstractITensorNetwork, edge::AbstractEdge)
+  ls = linkinds(tn, edge)
+  return prod([isnothing(l) ? 1 : dim(l) for l in ls])
+end
+
+function linkdims(tn::AbstractITensorNetwork)
+  return Dictionary(edges(tn), map(e -> linkdim(tn, e), edges(tn)))
```

Comment:
I'm going back and forth about whether or not this should output a `DataGraph` with the dimensions as metadata on the edges.

I think an advantage is that in principle it could more easily be used to then construct a new `ITensorNetwork` (kind of like how `IndsNetwork` is currently used, but more like an `IndexSpaceNetwork` or `SpaceNetwork`). Also, it should then automatically handle making sure that inputs of edges in either direction still output the correct dimension (i.e. `linkdims(tn)[(1, 1) => (2, 1)]` and `linkdims(tn)[(2, 1) => (1, 1)]` should both work and output the same dimension).

Reply:
I hadn't thought of indexing with edges in either direction (or with pairs of vertices for that matter), I'll change it to return a `DataGraph`. To be specific, I guess it would be a `NamedDimDataGraph` to be consistent with with the `data_graph` of an `ITensorNetwork`?

Reply:
Done, but just printing `linkdims` of a network is a bit weird now since it shows both edge orientations separately for each edge when printing a `DataGraph`. But that's a potential issue for DataGraphs.jl

Reply:
This should all be easier now with the latest updates to `DataGraphs.jl`, you should be able to just use a `DataGraph`.

Reply:
Done.